apps/confidential/confidential.go

```diff
@@ -156,11 +156,20 @@ func NewCredFromSecret(secret string) (Credential, error) {
 }
 
 // NewCredFromAssertion creates a Credential from a signed assertion.
+//
+// Deprecated: a Credential created by this function can't refresh the
+// assertion when it expires. Use NewCredFromAssertionCallback instead.
 func NewCredFromAssertion(assertion string) (Credential, error) {
 	if assertion == "" {
 		return Credential{}, errors.New("assertion can't be empty string")
 	}
-	return Credential{assertion: assertion}, nil
+	return NewCredFromAssertionCallback(func(context.Context) (string, error) { return assertion, nil }), nil
+}
+
+// NewCredFromAssertionCallback creates a Credential that invokes a callback to get assertions
+// authenticating the application.
+func NewCredFromAssertionCallback(callback func(context.Context) (string, error)) Credential {
+	return Credential{assertionCallback: callback}
```

Comment:
Overall, this PR's introduction of an assert callback looks good to me. My Golang skills (or the lack thereof) does not see a thread safety issue in this line, so I'll leave that part to the experts.